SU2_PY/SU2/eval/design.py

```diff
@@ -226,25 +226,64 @@ def obj_f(dvs,config,state=None):
     
     def_objs = config['OPT_OBJECTIVE']
     objectives = def_objs.keys()
-    
-#    if objectives: print('Evaluate Objectives')
+
     # evaluate each objective
     vals_out = []
     func = 0.0
     for i_obj,this_obj in enumerate(objectives):
         scale = def_objs[this_obj]['SCALE']
         sign  = su2io.get_objectiveSign(this_obj)
-        
         # Evaluate Objective Function
         # scaling and sign
-        func += su2func(this_obj,config,state) * sign * scale
-        
+        if def_objs[this_obj]['CTYPE']=='NONE':
+            func += su2func(this_obj,config,state) * sign * scale
+        else:
+            func += obj_p(config,state,this_obj,def_objs) * scale
     vals_out.append(func)
     #: for each objective
-    
+    # If evaluating the combined function is desired, update it here
+    if state.FUNCTIONS.has_key('COMBO'):
+        state['FUNCTIONS']['COMBO'] = func
+        
     return vals_out
 
 #: def obj_f()
+        
+def obj_p(config,state,this_obj,def_objs):
```

Comment:
I like the possibility of adding penalty functions... sometimes is quite difficult to make the constraint optimizer works.